TFNetworkLayer.py

```diff
@@ -8273,6 +8275,68 @@ def __init__(self,
     self.sampler_args = sampler_args
     if self.sampler_args is None:
       self.sampler_args = {}
+    self.log_norm_factor = log_norm_factor
+
+  def _nce_loss(self,
+                weights,
+                biases,
+                labels,
+                inputs,
+                num_sampled,
+                num_classes,
+                num_true=1,
+                sampled_values=None,
+                remove_accidental_hits=False,
+                partition_strategy="div",
+                name=None,
+                seed=None):
+    """
+    Returns the example-wise NCE losses for the batch.
+
+    This is mostly a copy of
+      https://github.com/tensorflow/tensorflow/blob/e19c354920c3b246dda6598229210a582caaa1a9/tensorflow/python/ops/nn_impl.py#L1729-L1837
+
+    :param tf.Tensor weights: NCE parameters of shape [num_classes, dim].
+    :param tf.Tensor biases: biases with shape [num_classes]
+    :param tf.Tensor labels: The target classes of shape [batch_size, num_true]
+    :param tf.Tensor inputs: The forward activations of the network of shape [batch_size, dim]
+    :param int num_sampled: Number of sampled classes
+    :param int num_classes: Number of possible classes
+    :param int num_true: Number of target classes per training example
+    :param (tf.Tensor[int64], tf.Tensor, tf.Tensor) sampled_values: a tuple of 
```

Comment:
I think `tf.Tensor[int64]` is invalid syntax. It should just be `tf.Tensor`. Write in the description that `sampled_candidates` is of dtype int64.